### PR TITLE
Fix: Modal and screen closing with single `esacpe` keypress

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ https://github.com/Expensify/App/issues/5769

### Tests | QA Steps

1. Navigate to Workspace settings on the web.
2. Navigate to members
3. Check one from the list of members and click on the remove button.
4. Once the remove modal appears, press the `esc` button.
5. Only Confirm modal should close.
6. Press Esc again.
7. Now the Settings Modal should close.

### Tested On

- [x] Web
- [ ] Mobile Web [No Escape key]
- [x] Desktop
- [ ] iOS [No Escape key]
- [ ] Android [No Escape key]

#### Web | Desktop

https://user-images.githubusercontent.com/24370807/137516779-ce4417d9-6607-47a6-9382-b9481e6bb942.mp4

### CODE REVIEW

File: src/components/ScreenWrapper.js [hunk 1]
```diff
@@ -34,6 +37,13 @@ const propTypes = {
         // Method to attach listener to Navigation state.
         addListener: PropTypes.func.isRequired,
     }),
+
+    /** Details about any modals being used */
+    modal: PropTypes.shape({
+        /** Indicates if there is a modal currently visible or not */
```
Comment (deetergp) [inline src/components/ScreenWrapper.js:43]: NAB: Your property names are pretty self-describing, which is great, so the doc comment is unnecessary.

File: src/components/ScreenWrapper.js [hunk 2]
```diff
@@ -116,4 +129,11 @@ class ScreenWrapper extends React.Component {
 ScreenWrapper.propTypes = propTypes;
 ScreenWrapper.defaultProps = defaultProps;
 ScreenWrapper.displayName = 'ScreenWrapper';
-export default withNavigation(ScreenWrapper);
+export default compose(
+    withNavigation,
+    withOnyx({
+        modal: {
```
Comment (deetergp) [inline src/components/ScreenWrapper.js:135]: NAB: Why does the modal state need to be in Onyx? Does't the modal's visibility only really matter to the client?

**Review by deetergp — DISMISSED** [review]
Looks good and tests pass 👍

File: src/components/ScreenWrapper.js (hunk 2 above)
Comment (parasharrajat) [reply src/components/ScreenWrapper.js:135]: Yup. You are correct but App is architected this way. and we only use Onyx for data flow.

**Review by Julesssss — APPROVED** [review]

**Review by Julesssss — CHANGES_REQUESTED** [review]
I'm not sure the exact reproduction steps, but I am still able to trigger the Esc bug occasionally. It's shown at the end of this clip, I close the confirmation dialog a couple of times, then hit escape which closes both the modal and screen.  

https://user-images.githubusercontent.com/10736861/137723793-b0afde2c-c42c-415f-be2c-b421723964f7.mov

**Review by Julesssss — APPROVED** [review]
Tests well 👍

**Comment by s77rt** [comment]
This PR could have prevented this bug https://github.com/Expensify/App/issues/18558. The `willAlertModalBecomeVisible` call is only made on `componentDidUpdate`. We should have made this call on `componentDidMount` as well to cover the cases where the modal is initially rendered visible.

**Pull request merged**